pages/settings.js

```diff
@@ -0,0 +1,380 @@
+    const [languageProfile, setLanguageProfile] = useState(null);
+    const [dataBot, setDataBot] = useState(null);
+    const [botToken, setBotToken] = useState('');
+    const [helloMessage, setHelloMessage] = useState('');
+    const [like, setLike] = useState('');
+    const [disLike, setDisLike] = useState('');
+    const [report, setReport] = useState('');
+    const [message, setMessage] = useState('');
+    const [accountSettings, setAccountSettings] = useState('');
+    const [menu, setMenu] = useState('');
+    const [back, setBack] = useState('');
+    const [startSearch, setStartSearch] = useState('');
+    const [editForm, setEditForm] = useState('');
+    const [restartForm, setRestartForm] = useState('');
+    const [editFormField, setEditFormField] = useState('');
+    const [activateAccount, setActivateAccount] = useState('');
+    const [deactivateAccount, setDeactivateAccount] = useState('');
+    const [yes, setYes] = useState('');
+    const [no, setNo] = useState('');
+    const [errorMessage, setErrorMessage] = useState('');
+    
+    useEffect(() => {
+        fetch('http://localhost:3000/api/botSettings/get')
+            .then(response => response.json())
+            .then(data => {
+                setDataBot(data);
+            })
+            .catch(error => {
+                console.error('Error:', error);
+            });
+    }, []);
+
+    useEffect(() => {
+        if (dataBot) {
+            fetch('http://localhost:3000/api/languageProfile/one/1')
+                .then(response => response.json())
+                .then(data => {
+                    setLanguageProfile(data);
+                })
+                .catch(error => {
+                console.error('Error:', error);
+                });
+        }
+    }, [dataBot]);
+    useEffect(() => {
+        if (dataBot && languageProfile){
+            setBotToken(dataBot.botToken)
+            setHelloMessage(languageProfile.helloMessage)
+            setLike(dataBot.like)
+            setDisLike(dataBot.disLike)
+            setReport(dataBot.report)
+            setMessage(dataBot.message)
+            setAccountSettings(languageProfile.buttonNames.accountSettings)
+            setMenu(languageProfile.buttonNames.menu)
+            setBack(languageProfile.buttonNames.back)
+            setStartSearch(languageProfile.buttonNames.startSearch)
+            setEditForm(languageProfile.buttonNames.editForm)
+            setRestartForm(languageProfile.buttonNames.restartForm)
+            setEditFormField(languageProfile.buttonNames.editFormField)
+            setActivateAccount(languageProfile.buttonNames.activateAccount)
+            setDeactivateAccount(languageProfile.buttonNames.deactivateAccount)
+            setYes(languageProfile.buttonNames.yes)
+            setNo(languageProfile.buttonNames.no)
+        }
+    }, [dataBot, languageProfile]);
+
+    const handleSubmit = (e) => {
+        e.preventDefault();
+
+        if (botToken === '') {
```

Comment:
I suggest using formik to help working with forms, especially its validation - https://formik.org/docs/guides/validation

Reply:
fixed